wp-content/themes/twentytwenty-child/functions.php

```diff
@@ -557,8 +557,54 @@ function search_oxford_dictionary(){
 /** 
  * Creating shortcode for students
  * */
-function students_shortcode() {
+function students_shortcode( $attributes ) {
+    $shortcode_args = shortcode_atts( array(
+        'number_of_students' => '0',
+    ), $attributes );
 
+    ?>
+
+    <div id="site-content" class="<?php echo is_page_template( 'templates/template-full-width.php' ) ? '' : 'thin'; ?>">
+    <div id="content" role="main" class="archive-header-inner section-inner medium">
+ 
+    <?php 
+    $args = array(
+        'post_type'      => 'student',
+        'meta_key'       => 'status',
+        'meta_value'     => '1',
+        'posts_per_page' => $shortcode_args[ 'number_of_students' ]
+    );
+    $query = new WP_Query( $args );
+    
+    if( $query->have_posts() ) {
+        while ( $query->have_posts() ) {
+    ?>
+
+    <div class="student-box"> 
+
+        <?php
+            $query->the_post();
+
+            $title = str_replace( ' ','-', strtolower( get_the_title() ) );
+            $data = get_student_info( get_the_ID() );
+
+            echo '<div class="student-name"> <a href=' . home_url( $title ) . '>' . get_the_title() . ', ' . $data[ 'class' ] . ' Grade </a> </div>';
```

Comment:
Again you should use `get_permalink()`. Aaaand, you should always escape your data before outputting it.

In this case here, you can use `esc_html()`.

Reply:
Escaping is even valid for when you using built-in functions like `get_the_title()` and `get_permalink`. Even though they come with WordPress you should still escape them.